test/org/sagebionetworks/bridge/validators/StudyParticipantValidatorTest.java

```diff
@@ -94,122 +102,200 @@ public void validatesUpdate() {
+        validator = new StudyParticipantValidator(externalIdService, study, false);
+        study.setExternalIdValidationEnabled(true);
+        
+        StudyParticipant participant = new StudyParticipant.Builder().withId("id").withEmail("email@email.com")
+                .withPassword("aAz1%_aAz1%").withExternalId("foo").build();
+        
+        Validate.entityThrowingException(validator, participant);
+    }
+    
+    @Test
+    public void validateExistingManagedExternalId() {
+        when(externalIdService.getExternalId(study.getStudyIdentifier(), "foo")).thenReturn(EXT_ID);
+
+        StudyParticipant participant = new StudyParticipant.Builder().withEmail("email@email.com")
+                .withExternalId("foo").withId("id").withPassword("pAssword1@").build();
+        
+        validator = new StudyParticipantValidator(externalIdService, study, false);
+        study.setExternalIdValidationEnabled(true);
+        Validate.entityThrowingException(validator, participant);
+    }
+    
+    @Test
+    public void externalIdManagedButInvalidOnUpdateOK() {
+        StudyParticipant participant = new StudyParticipant.Builder().withEmail("email@email.com")
+                .withExternalId("foo").withId("id").withPassword("pAssword1@").build();
+        
+        // The ID won't exist and that's fine, because you cannot add an external ID on an update.
+        // We're just going to ignore email, phone, and external ID on updates.
+        validator = new StudyParticipantValidator(externalIdService, study, false);
+        study.setExternalIdValidationEnabled(true);
+        Validate.entityThrowingException(validator, participant);
+    }
+    
+    @Test
+    public void externalIdValidWithoutManagement() {
+        StudyParticipant participant = new StudyParticipant.Builder().withEmail("email@email.com")
+                .withExternalId("foo").withId("id").withPassword("pAssword1@").build();
+        
+        // The ID won't exist and that's fine
+        validator = new StudyParticipantValidator(externalIdService, study, true);
+        study.setExternalIdValidationEnabled(false);
+        study.setExternalIdRequiredOnSignup(true);
+        Validate.entityThrowingException(validator, participant);
+    }
+    
+    @Test
+    public void externalIdInvalidWithManagement() {
+        when(externalIdService.getExternalId(study.getStudyIdentifier(), "foo")).thenReturn(EXT_ID);
+        
+        StudyParticipant participant = new StudyParticipant.Builder().withEmail("email@email.com")
+                .withId("id").withPassword("pAssword1@").build();
+        
+        // This should fail. External ID is required on sign up but there's no externalId
+        validator = new StudyParticipantValidator(externalIdService, study, true);
+        study.setExternalIdValidationEnabled(true);
+        study.setExternalIdRequiredOnSignup(true);
+        assertValidatorMessage(validator, participant, "externalId", "is required");
+    }
```

Comment:
I think we're still missing a test case. If external ID validation is enabled, but no external ID is specified, the validation would succeed. However, there is no test that tests this case.